Esther125/CloudDrop
# Feature/yun si/cns 7 project init

# Change
最後還是使用前後端分離的架構，分別放在 frontend, backend 兩個資料夾中

### Backend
- Add sampleRoutes
- Add SampleController with greet method
- Add Dependencies
  - **nodemon** for hot-reload
  - **body-parser, cors** to handle request and response

### Frontend
- call http://localhost:3000/api/v1/sample/greet this api in `components/HelloWorld.vue`

# Test Steps

1. follow the doc (`docs/project_setup/project_init.md`) to install needed dependencies and start project

backend console:
```sh
> backend@1.0.0 start
> nodemon --env-file=.env index.js

[nodemon] 3.1.7
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,cjs,json
[nodemon] starting `node --env-file=.env index.js`
Server is running on port 3000
```

frontend console:
```sh
> frontend@0.0.0 dev
> vite


  VITE v5.4.9  ready in 150 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: use --host to expose
  ➜  press h + enter to show help
```
ps. port may be different

2. click frontend url: http://localhost:5173/

Then you will see:
```
Response Status: 200
Response Body: { "message": "hi" }
```

![image](https://github.com/user-attachments/assets/3cf22332-b1bf-4dc4-a147-d64491e95f4f)
